skymp5-functions-lib/src/logic/SweetPieGameModeListener.ts

```diff
@@ -47,11 +47,24 @@ export class SweetPieGameModeListener implements GameModeListener {
   constructor(private controller: PlayerController, private maps: SweetPieMap[] = [], private minimumPlayersToStart: number = 5) {
     this.rounds = this.controller.getRoundsArray();
     if (this.rounds.length === 0) {
-      maps.forEach(map => this.rounds.push({ state: 'wait', map: map }));
-      this.rounds.forEach((round, index) => this.resetRound(index));
+      maps.forEach(map => this.rounds.push({ state: 'wait', map: map, hallPointName: this.hallSpawnPointName, secondsPassed: 0 }));
     } else {
-      maps.forEach((map, index) => this.rounds[index].map = map);
+      if (maps.length > this.rounds.length) {
```

Comment:
It's complicated. We probably should just reset all rounds on any maps change. Including ids and spawn points changes. It will be much more predictable.

Reply:
If there're any players in round, it will be reset after round ends